src/taipy/core/notification/notifier.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2023 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+from queue import Queue
+from typing import List, Tuple
+
+from .event import Event
+from .registration import Registration
+from .topic import Topic
+
+
+class Notifier:
+    _registrations: List[Registration] = []
+
+    @classmethod
+    def register(cls, entity_type, entity_id, operation, attribute_name) -> Tuple[str, Queue]:
```

Comment:
The parameter is really specific, but what if I want to register an entity with all of its operation? Do I have to do it multiple times? Instead of entity_type, entity_id, I think can ask the user to just provide the entity. What do you think?

Reply:
I suggest making all the parameters optional. Like:

- SCENARIO, None, None, None -> Get notified for any operation on any scenario
- SCENARIO, None, UPDATE, None -> Get notified for any UPDATE operation on any scenario
- None, ID, UPDATE, DESCRIPTION -> Get notified for any UPDATE operation on attribute DESCRIPTION of any entity with id ID

Reply:
So for a registration, we can have multiple topics yes?